### PR TITLE
render_hud_element

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.07%) on `us`.

🚀 This PR matches 1 function (+0.07%) on `ique`.

File: src/hud_element.c [hunk 1]
```diff
@@ -1517,19 +1517,19 @@ void render_hud_element(HudElement* hudElement) {
                 }
             } else {
                 if (hudElement->flags & HUD_ELEMENT_FLAG_TRANSPARENT) {
-                    fold_update(transform->foldIdx, FOLD_TYPE_7, 255, 255, 255, hudElement->opacity, 0);
+                    fold_update(transform->foldIdx, 7, 255, 255, 255, hudElement->opacity, 0);
```
Comment (HailSanta) [inline src/hud_element.c:1520]: missing the fold enums here

File: src/hud_element.c [hunk 2]
```diff
@@ -1420,18 +1418,17 @@ void render_hud_element(HudElement* hudElement) {
...
     FoldImageRecPart sp2A0;
-    f32 phi_f24, phi_f20;
+    f32 xScaleFactor, yScaleFactor;
     s32 height, width;
     HudTransform* transform;
     s32 mode;
     u32 flags;
     u8* raster;
-    u8* palette;
+    u16* palette;
```
Comment (HailSanta) [inline src/hud_element.c:1427]: PAL_PTR

File: src/hud_element.c [hunk 3]
```diff
@@ -1420,18 +1418,17 @@ void render_hud_element(HudElement* hudElement) {
     Matrix4f sp220;
     Matrix4f sp260;
     FoldImageRecPart sp2A0;
-    f32 phi_f24, phi_f20;
+    f32 xScaleFactor, yScaleFactor;
     s32 height, width;
     HudTransform* transform;
     s32 mode;
     u32 flags;
     u8* raster;
```
Comment (HailSanta) [inline src/hud_element.c:1426]: IMG_PTR

File: src/hud_element.c [hunk 4]
```diff
@@ -276,15 +276,15 @@ void hud_element_draw_rect(HudElement* hudElement, s16 texSizeX, s16 texSizeY, s
     s32 masks, maskt;
     s32 screenPosOffsetScaledX, screenPosOffsetScaledY;
     u8* imageAddr;
-    u8* paletteAddr;
+    u16* paletteAddr;
```
Comment (HailSanta) [inline src/hud_element.c:279]: IMG_PTR / PAL_PTR

**Review by HailSanta — APPROVED** [review]

**Pull request merged**